listenbrainz/db/dump.py

```diff
@@ -301,6 +349,96 @@ def create_public_dump(location, dump_time, threads=DUMP_DEFAULT_THREAD_COUNT):
     )
 
 
+def create_feedback_dump(location, dump_time, threads=DUMP_DEFAULT_THREAD_COUNT):
+    """ Create a spark format dump of user listen and user recommendation feedback.
+    """
+    return _create_dump(
+        location=location,
+        dump_type='feedback',
+        tables=[],
+        dump_time=dump_time,
+        threads=threads,
+    )
+
+
+def dump_user_feedback(connection, location):
+    """ Carry out the actual dumping of user listen and user recommendation feedback.
+    """
+
+    with connection.begin() as transaction:
+
+        # First dump the user feedback
+        result = connection.execute(sqlalchemy.text("""
+            SELECT musicbrainz_id, recording_msid, score, r.created,
+                   EXTRACT(YEAR FROM r.created) AS year,
+                   EXTRACT(MONTH FROM r.created) AS month,
+                   EXTRACT(DAY FROM r.created) AS day
+              FROM recording_feedback r
+              JOIN "user"
+                ON r.user_id = "user".id
+          ORDER BY created"""))
+
+        last_day = ()
+        todays_items = []
+
+        while True:
```

Comment:
There's some scope of de-duplicating the code here. The `while` loop can probably be extraced into a separate method of its own receiving `key` names like `recording_msid` or `mb_recording_id` so on.

Reply:
Yeah, I see what you mean, but doing so would complicate the code and make it less readable in order to avoid duplicating a few lines of code. I would much prefer to keep it as is.